persistence/src/main/java/cz/fi/muni/pa165/model/dao/UserDaoImpl.java

```diff
@@ -44,11 +48,25 @@ public User findByName(PersonName name) {
         try{
             return q.setParameter("name", name).getSingleResult();
         } catch (NoResultException nre) {
+            log.warn("no user found with name " + name);
+            log.warn("returning null");
             return null; // todo: really? or exception instead? Should be consistent with other methods.
         }
     }
 
     @Override
+    public User findByEmail(String email) {
+        try {
+            return em.createQuery("SELECT u FROM User u WHERE u.email = :email", User.class)
+                    .setParameter("email", email).getSingleResult();
+        } catch (NoResultException nre){
+            log.warn("no user found with email " + email);
+            log.warn("returning null");
+            return null;
```

Comment:
Try to use
`mvnDebug tomcat7:run` instead. In Idea then set-up remote debuggin session. (Remote run configuration) And you can debug thing just like if they are run from Idea. Cool, right?